cdap-ui/app/features/mapreduce/controllers/tabs/status-ctrl.js

```diff
@@ -1 +1,85 @@
-// Once the status page has enough content this controller should be used.
+angular.module(PKG.name + '.feature.mapreduce')
+  .controller('MapreduceStatusController', function($scope, MyDataSource, $state, $rootScope) {
+    var dataSrc = new MyDataSource($scope),
+        basePath = '/apps/' +
+            $state.params.appId +
+            '/mapreduce/' +
+            $state.params.programId;
+
+    dataSrc.request({
+      _cdapNsPath: basePath + '/runs'
+    })
+    .then(function(runs) {
+      $scope.latestRun = runs[0];
+
+      dataSrc.poll({
+        _cdapNsPath: '/apps/' + $state.params.appId
+                      + '/mapreduce/' + $state.params.programId
+                      + '/runs/' + runs[0].runid + '/info'
+      }, function (res) {
+
+        $scope.info = res;
+        $scope.mapProgress = Math.floor(res.mapProgress * 100);
+        $scope.reduceProgress = Math.floor(res.reduceProgress * 100);
+
+        $scope.mapperStats = getStats($scope.info.mapTasks);
+        $scope.reducerStats = getStats($scope.info.reduceTasks);
+      });
+
+
+
+    });
+
+
+    $scope.getCompletedPercentage = function(tasks) {
+      var aggregate = 0;
+      angular.forEach(tasks, function (task) {
+        if(task.state === 'SUCCEEDED') {
+          aggregate += task.progress;
+        }
+      });
+      return aggregate ? ((aggregate / (tasks.length)) * 100).toFixed(1) : 0;
+    };
+
+    $scope.getFailedTasks = function (tasks) {
```

Comment:
This seems redundant to the computation below (lines 55-80), where it computes the stats.failed there also.
Or is there a difference?